### ISSUE TITLE
Extend Temporal controller with all numeric data types

### ISSUE THREAD

**Issue by jmonticolo** [qgis/QGIS#40767]
**Feature description**
Temporal controller was a huge new feature for QGIS for see spatial data through time. But imagine a vector layer with a numeric depth / height field (or any kind of numerical data type) that we could vary with the slider control ? The request is for extend the slider not just for time data, but for all numerical fields.

**Additional context**
As temporal controller, in the vector layer properties, we can select the field, the data type (time or numeric), the start and the end values.

**Comment by roya0045** [comment]
You could convert the numeric field to a date + the field value as offset.

**Comment by jmonticolo** [comment]
@roya0045 : yes, you are totally right, I have thought about this possibility but this feature request is for make things more clear and less tricky for the user.

**Comment by rduivenvoorde** [comment]
@jmonticolo it is a deliberate choice to ONLY use QDateTime fields as time formats. And IF an attribute is indeed time, then you can create a 'virtual' column for it (using an expression to convert your string/int to a date). That works great.

BUT I think you are talking about a more general 'dimension' controller (like in Geoserver you can either use WMS-T on time or on other (sortable) dimensions like depth etc. To get there we really need another Controller I think. Current one is very dedicated to time use...

**Comment by jmonticolo** [comment]
@rduivenvoorde yes, a more general 'dimension' controller is exactly what I am thinking.

**Comment by kannes** [comment]
This is a duplicate of https://github.com/qgis/QGIS/issues/39804 I think
